timestamp-impl/src/main/java/com/palantir/timestamp/PersistentTimestampServiceImpl.java

```diff
@@ -71,15 +72,16 @@ public static PersistentTimestampService create(ErrorCheckingTimestampBoundStore
     }
 
     @VisibleForTesting
-    PersistentTimestampServiceImpl(PersistentTimestamp timestamp) {
+    PersistentTimestampServiceImpl(ErrorCheckingTimestampBoundStore store, PersistentTimestamp timestamp) {
+        this(store);
         this.timestamp = timestamp;
     }
 
     private PersistentTimestampServiceImpl(ErrorCheckingTimestampBoundStore store) {
         this.store = store;
     }
 
-    private void tryInitialize() {
+    private synchronized void tryInitialize() {
```

Comment:
maybe missing something: I get that the `timestamp` reference not being volatile in the past was a risk that other threads may not see the update here, though do we need to both make it volatile and make this synchronized?

Reply:
I want to ensure we serialize initialization to ensure safety, as we should only be doing this at service startup